content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,994 @@
+Here are some things to keep in mind when building a Composition Function:
+
+* Your Function may be running as part of a pipeline. This means your Function
+  _must_ pass through any desired state that it's unconcerned with. If your
+  Function is passed a desired composed resource and doesn't return that
+  composed resource in its output, it will be deleted. Crossplane considers the
+  desired state of the XR and any composed resources to be whatever `FunctionIO`
+  is returned by the last Function in the pipeline.
+* Crossplane won't set a `metadata.name` for your desired resources resources.
+  It's a good practice to match P&T Composition's behavior by setting
+  `metadata.generateName: "name-of-the-xr-"` for any new desired resources.
+* Don't add new entries to the desired resources array every time your function
+  is invoked. Remember to check whether your desired resource is already in the
+  `observed` and/or `desired` objects. You may need to update it rather than
+  create it.
+* Don't bypass providers. Composition Functions are designed to tell Crossplane
+  how to orchestrate managed resources - not to directly orchestrate external
+  systems.
+* Include your function name and version in any results you return to aid in
+  debugging.
+* Write tests for your function. Pass it a `FunctionIO` on stdin in and ensure
+  it returns the expected `FunctionIO` on stdout.
+* Keep your Functions fast and lightweight. Remember that Crossplane runs them
+  approximately once every 30-60 seconds.
+
+## The xfn runner
+
+Composition Function runners are designed to be pluggable. Each time Crossplane
+needs to invoke a Composition Function it makes a gRPC call to a configurable
+endpoint. The default, reference Composition Function runner is named `xfn`.
+
+{{< hint "note" >}}
+The default runner endpoint is `unix-abstract:crossplane/fn/default.sock`. It's
+possible to run Functions using a different endpoint, for example:
+
+```yaml
+  functions:
+  - name: my-cool-Function
+    type: Container
+    container:
+      image: xkpg.io/my-cool-Function:0.1.0
+      runner:
+        endpoint: unix-abstract:/your/custom/runner.sock
+```
+
+Currently Crossplane uses unauthenticated, unencrypted gRPC requests to run
+Functions, so requests shouldn't be sent over the network. Encryption and
+authentication will be added in a future release.
+{{< /hint >}}
+
+`xfn` runs as a sidecar container within the Crossplane pod. It runs each
+Composition Function as a nested [rootless container][rootless-containers].
+
+{{< img src="master/guides/composition-functions-xfn-runner.png" alt="Crossplane running Functions using xfn via gRPC" size="tiny" >}}
+
+The Crossplane Helm chart deploys `xfn` with:
+
+* The [`Unconfined` seccomp profile][kubernetes-seccomp].
+* The `CAP_SETUID` and `CAP_SETGID` capabilities.
+
+The `Unconfined` seccomp profile allows Crossplane to make required syscalls
+such as `unshare` and `mount` that are not allowed by most `RuntimeDefault`
```

Comment:
>to make required syscalls
>such as `unshare` and `mount`

Are these the only required syscalls? We should either list the required syscalls or remove the "such as unshare and mount".  Either be precise "to make the following syscalls" or abstract "to make the required syscalls [end]"

Reply:
>that are not allowed by most `RuntimeDefault` profiles.

By saying "most" and "RuntimeDefault profiles" it's creating an assumption about seccomp we need to clarify. Something like "allowed by most _container runtime default seccomp profiles_". 

How do we indicate to the reader where they should go look to find out if they are in the "most" camp or not.

Reply:
I don't know the exact set of syscalls that are prevented by the RuntimeDefault implementations. It's probably possible to give the reader a better idea, but I'm not sure how valuable that is in this page of documentation.

Part of the challenge here is that the `RuntimeDefault` policy means "use the default policy of your OCI runtime", and that default policy changes from runtime to runtime so it's hard to definitively say what the difference will be.

Reply:
I would suggest removing the "such as..." part then. Mentioning some without saying what is required creates the question "well, what else?". If it's not mentioned then a reader is less likely to question.